tap_googleads/client.py

```diff
@@ -78,6 +79,54 @@ def get_url_params(
             params["sort"] = "asc"
             params["order_by"] = self.replication_key
         return params
+    
+    def validate_response(self, response):
+        # Still catch error status codes
+        if response.status_code == 403:
+            msg = (
+                f"{response.status_code} Client Error: "
+                f"{response.reason} for url: {response.url}"
+            )
+            data = response.json()
+            details: dict = data.get("error").get("details")
+            
+            if details and details[0]["errors"][0]["errorCode"]["authorizationError"] == "CUSTOMER_NOT_ENABLED":
+                raise CustomerNotEnabledError(msg)
+            
+        if 400 <= response.status_code < 500:
+            msg = (
+                f"{response.status_code} Client Error: "
+                f"{response.reason} for path: {self.path}"
+            )
+            raise FatalAPIError(msg)
+
+        elif 500 <= response.status_code < 600:
+            msg = (
+                f"{response.status_code} Server Error: "
+                f"{response.reason} for path: {self.path}"
+            )
+            raise RetriableAPIError(msg)
+    
+    def get_records(self, context: Optional[dict]) -> Iterable[Dict[str, Any]]:
+        """Return a generator of row-type dictionary objects.
+
+        Each row emitted should be a dictionary of property names to their values.
+
+        Args:
+            context: Stream partition or context dictionary.
+
+        Yields:
+            One item per (possibly processed) record in the API.
+        """
+        try:
+            for record in self.request_records(context):
+                transformed_record = self.post_process(record, context)
```

Comment:
The post_process function in the client just returns the record. Are these lines needed, or should we delete post_process?

Reply:
Right now you're right we don't need the post_process method.

My thought was to just do what the SDK is doing by default in https://gitlab.com/meltano/sdk/-/blob/main/singer_sdk/streams/rest.py#L385-401

Idea being it's easy to come into this library as someone familiar with the SDK and write a post_process method if/when it's needed. Let's say a stream we add in the future needs some kind of post_processing on one of the fields for example

Reply:
To go a bit deeper, handling a "warning" instead of a failure is something that should probably be an sdk feature. (So I can see the argument for simplifying this method as much as possible which is fine, but I think I'd rather leave it)

There's an issue into gitlab here https://gitlab.com/meltano/sdk/-/issues/134 which covers the main issue of when should you allow a "partial failure" run to happen and how should that be handled by Meltano.